test/extend_test.jl

```diff
@@ -20,8 +20,8 @@ d[1, 1:5] = [1.1231, 1.313, 5.123, 2.231, 4.1231]
 set_dims!(d, (1, 5))
 @test size(d) == (1, 5)
 
-@test d[:, end] ≈ [4.1231]
-@test d[end, :] == [1.1231 1.313 5.123 2.231 4.1231]
+@test d[:, end] ≈ 4.1231
```

Comment:
I think this should still be an array... only indexing with an Integer drops dims:

```julia
julia> rand(1, 5)[:, end]
1-element Array{Float64,1}:
 0.4666714180536573

julia> rand(1, 5)[1, end]
0.013875531617609127
```

I can't find anything in the array interface docs that is really clear on this, just copying Base indexing is probably best. Maybe we need a "taking the array interface seriously" discussion on discourse

Reply:
Please feel free to open a PR if you have some ideas

Reply:
Ok I'll try, it might be a month or two with my current workload